source/IpcServer.cpp

```diff
@@ -0,0 +1,102 @@
+#include "IpcServer.hpp"
+#include "Server.hpp"
+#include "Output.hpp"
+
+#include <iostream>
+#include <unistd.h>
+#include <signal.h>
+
+IpcServer::IpcServer(std::string const& socket, Server *server)
+  : ipcServer(socket)
+  , socket(socket)
+  , server(server)
+  , clients()
+  , acceptThread(&IpcServer::acceptClients, this)
+  , processThread(&IpcServer::processClients, this)
+{
+  // Ignore broken pipe
+  signal(SIGPIPE, SIG_IGN);
+}
+
+void IpcServer::acceptClients()
+{
+  while (1)
+    {
+      try {
+        std::unique_ptr<libsocket::unix_stream_client> client = ipcServer.accept2();
+        std::cout << "New ipc client" << std::endl;
+        clients.emplace_back(std::move(client));
+
+      } catch (const libsocket::socket_exception& e) {
+        std::cerr << e.mesg;
+      }
+    }
+}
+
+void IpcServer::processClients()
+{
+  int size = -1;
+  int activeWorkspaceId = -1;
+  unsigned int clientSize = -1;
+  while (1)
+    {
+      usleep(100);
```

Comment:
I'd rather have a blocking solution, but since we're under time pressure I'm ok merging as-is.